src/protocols/secure_channel/tests/TestCheckinMsg.cpp

```diff
@@ -0,0 +1,267 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <protocols/secure_channel/CheckinMessage.h>
+
+#include <crypto/DefaultSessionKeystore.h>
+#include <lib/support/BufferWriter.h>
+#include <lib/support/CHIPMem.h>
+#include <lib/support/UnitTestRegistration.h>
+#include <protocols/Protocols.h>
+#include <protocols/secure_channel/Constants.h>
+#include <protocols/secure_channel/StatusReport.h>
+#include <system/SystemPacketBuffer.h>
+
+#include <crypto/tests/AES_CCM_128_test_vectors.h>
+
+#include <lib/support/UnitTestExtendedAssertions.h>
+#include <lib/support/UnitTestRegistration.h>
+#include <nlunit-test.h>
+
+using namespace chip;
+using namespace chip::Protocols;
+using namespace chip::Protocols::SecureChannel;
+using TestSessionKeystoreImpl = Crypto::DefaultSessionKeystore;
+
+void TestCheckin_Generate(nlTestSuite * inSuite, void * inContext)
```

Comment:
We should have a test the validates that inputs generate the correct output. In other words, validate set of bytes are correct.

Reply:
That is mostly handled by the Unit test of encryption and decryption. There is no point invalidating that theses API works or not since it should already be taken care of.